### PR TITLE
fix(proj-creation): Add toast for proj creation

### PR DESCRIPTION
**Pull request by schew2381**
Now include a toast for creating a project. If using the project creation flow, the toast will include info about the created team. Also create a failure toast on project failure.

### CODE REVIEW

File: src/sentry/api/endpoints/organization_projects_experiment.py [hunk 1]
```diff
@@ -188,5 +188,7 @@ def post(self, request: Request, organization: Organization) -> Response:
             "created team through project creation flow",
             extra={"team_slug": default_team_slug, "project_slug": project_name},
         )
+        serialized_response = serialize(project, request.user)
+        serialized_response["team_slug"] = team.slug
 
-        return Response(serialize(project, request.user), status=201)
+        return Response(serialized_response, status=201)
```
Comment (sentaur-athena) [inline src/sentry/api/endpoints/organization_projects_experiment.py:194]: what is the difference between serializing response and passing as text field. For example, why do we serialize project in response but not team?

File: static/app/views/projectInstall/createProject.tsx [hunk 2]
```diff
@@ -149,6 +149,23 @@ function CreateProject() {
...
+            tct('Created project [project]', {
+              project: `${projectData.slug}`,
+              organization: organization.slug,
+            })
+          );
+        } else {
+          addSuccessMessage(
+            tct('Created project [project] under new team [team]', {
+              project: `${projectData.slug}`,
+              team: `#${projectData.team_slug}`,
```
Comment (sentaur-athena) [inline static/app/views/projectInstall/createProject.tsx:163]: why is there a # before team slug and not project slug? Also why is there a # next to project slug in line 179

**Review by sentaur-athena — COMMENTED** [review]
Looks good to me. Can you help clarify these questions?

File: static/app/views/projectInstall/createProject.tsx (hunk 2 above)
Comment (schew2381) [reply static/app/views/projectInstall/createProject.tsx:162]: Only the team slug should have a `#` b/c that's how teams show up in the UI. Line 179 should not have the `#`, that was my mistake

File: src/sentry/api/endpoints/organization_projects_experiment.py (hunk 1 above)
Comment (schew2381) [reply src/sentry/api/endpoints/organization_projects_experiment.py:194]: We serialize the project in the response b/c the primary purpose of this API is to create a project. We need to add the `team_slug` in the resp as well b/c it's required in the toast we're displaying to the user. The toast only requires the `team_slug` so we don't need to serialize the rest of team

**Review by sentaur-athena — APPROVED** [review]
Looks great! Thanks for fixing the tests.

**Review by cathteng — APPROVED** [review]

**Pull request merged**